specifyweb/frontend/js_src/lib/components/DataModel/resource.ts

```diff
@@ -225,12 +225,37 @@ export function resourceOn(
 export const parseJavaClassName = (className: string): string =>
   className.split('.').at(-1) ?? '';
 
+const interactionTablesWithPreps = [
+  'Disposal',
+  'Loan',
+  'Gift',
+  'ExchangeIn',
+  'ExchangeOut',
+  'Borrow',
+] as const;
+type InteractionTable = typeof interactionTablesWithPreps[number];
+const interactionTablesPrepsFieldName: RR<InteractionTable, string> = {
+  Disposal: 'disposalPreparations',
+  Loan: 'loanPreparations',
+  Gift: 'giftPreparations',
+  ExchangeOut: 'exchangeOutPreps',
+  ExchangeIn: 'exchangeInPreps',
+  Borrow: 'borrowPreparations',
+};
```

Comment:
do you need to hardcode this list?
imagine someone adding a new table - how would they know that this place in the code might need to be modified before the table is added or else a bug is introduced???
that is why you should avoid hardcoding things and creating edge cases where possible - instead compute things on the fly

could you please dynamically construct it on the fly (i.e if field name is a relationship to the Preparations table, then don't clone it)

---

the only exception to computing things on the fly is when doing so is computationally expensive - but in those cases, you should add a test case that will do the computation to make sure the hardcoded list is not out of date with the dynamically computed list
(i.e front-end stores a hardcoed list of all the permission policies that sp7 back-end defines so as not to have to fetch those from the back-end on each app load - yet, when in development, it will make sure that the hardcoded list stored on the front-end matches exactly what back-end provides)

Reply:
The type `InteractionTable` already exists on the frontend (called `AnyInteractionPreparation`), and is constructed dynamically based on the `Tables` type. 

https://github.com/specify/specify7/blob/3fb636eaaec678d7ef4d2e175ac62f105b226cdb/specifyweb/frontend/js_src/lib/components/DataModel/helperTypes.ts#L58-L69

---

For defining a constant with this type, take a look at my changes in #4805. 
Specifically, `interactionPrepTables`: 

https://github.com/specify/specify7/blob/5638811529378ccbf1544e91d9b78c0ead81614c/specifyweb/frontend/js_src/lib/components/Interactions/helpers.ts#L7-L15

If needed/desired, there is also code to get the main Interaction tables which have preparations: 

https://github.com/specify/specify7/blob/5638811529378ccbf1544e91d9b78c0ead81614c/specifyweb/frontend/js_src/lib/components/Interactions/helpers.ts#L20-L26

--- 

#4805 also has examples of using/getting the relationship from Interaction -> InteractionPreparation: 
https://github.com/specify/specify7/blob/5638811529378ccbf1544e91d9b78c0ead81614c/specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx#L234-L241